File: src/main/java/dev/themeinerlp/plugindebug/DebugFileAdapter.java
package dev.themeinerlp.plugindebug;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;
import java.nio.file.Path;

final class DebugFileAdapter extends TypeAdapter<DebugFile> {
    @Override
    public void write(JsonWriter out, DebugFile value) throws IOException {
        out.beginObject()
                .name("filePath").value(value.filePath().getFileName().toString())
                .name("fileType").value(value.fileType().name())
                .name("uiTabName").value(value.uiTabName())
                .endObject();
    }

    @Override
    public DebugFile read(JsonReader in) throws IOException {
        in.beginObject();
        var filePath = in.nextString();
        var fileType = FileType.valueOf(in.nextString());
        var uiTabName = in.nextString();
        return new DebugFile(Path.of(filePath), fileType, uiTabName);
    }
}

File: src/main/kotlin/net/onelitefeather/plugindebug/DebugFileAdapter.kt
package net.onelitefeather.plugindebug

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import java.nio.file.Paths
import kotlin.io.path.name

internal class DebugFileAdapter : TypeAdapter<DebugFile>() {
    override fun write(out: JsonWriter, value: DebugFile) {
        out
            .beginObject()
            .name("filePath").value(value.filePath.name)
            .name("fileType").value(value.fileType.name)
            .name("uiTabName").value(value.uiTabName)
            .endObject()
    }

    override fun read(`in`: JsonReader): DebugFile {
        `in`.beginObject()
        val filePath = `in`.nextString()
        val fileType = FileType.valueOf(`in`.nextString())
        val uiTabName = `in`.nextString()
        `in`.endObject()
        return DebugFile(Paths.get(filePath), fileType, uiTabName)
    }
}